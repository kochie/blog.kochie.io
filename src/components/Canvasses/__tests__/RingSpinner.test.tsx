import React from 'react'
import { ReactTestRenderer, act, create } from 'react-test-renderer'
import Simulation from '@/components/Canvasses/ring-spinner'
// import { jest } from '@jest/globals'

// jest.createMockFromModule('d3')

describe('THEME COMPONENT', () => {
  test('should render', () => {
    let tree: ReactTestRenderer

    act(() => {
      tree = create(<Simulation />)
    })

    // @ts-expect-error tree will be assigned
    expect(tree.toJSON()).toMatchSnapshot()
  })
})
