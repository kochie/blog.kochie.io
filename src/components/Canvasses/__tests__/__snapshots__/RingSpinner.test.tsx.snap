// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`THEME COMPONENT should render 1`] = `
<div
  className="w-full my-10"
>
  <div
    className="flex gap-4 justify-center mb-6 mx-auto"
  >
    <input
      className="w-96"
      max="120"
      min="0"
      onChange={[Function]}
      type="range"
      value={5}
    />
    <div>
      5
      %
    </div>
    <div>
      0.02
       ms
      <sup>
        -2
      </sup>
    </div>
    <div>
      350.18
       ms
      <sup>
        -1
      </sup>
    </div>
  </div>
  <div
    className="flex flex-col md:flex-row w-full justify-evenly"
  >
    <canvas
      className="dark:bg-gray-800 rounded-2xl bg-gray-400"
      height={400}
      width={400}
    />
    <svg
      className="rounded-2xl dark:bg-gray-800 bg-gray-400"
      height={400}
      width={400}
    />
  </div>
</div>
`;
